131: update tile api

davidabrahams opened the pull request:
closes #131

This PR did some general code cleaning. It made finders return strings instead of ids. We then use `tileIsType` to actually find the location of those tiles, which takes in a string.

All other user-facing tile functions take in the `id`, so no conversion is needed by the user. The main functionality change is that if we ever pass an id that's not the type we have stored in our `tileInfo`, (ie, float when it should be string, or vice versa), the function will throw an error. This will force us to discover the correct id types for the tagpro tiles.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/chauncy-crib/tagprobot/188)
<!-- Reviewable:end -->

shanek21 commented:
Reviewed 7 of 8 files at r1, 1 of 1 files at r2.
Review status: 0 of 8 files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyhiO3KAWBQloZ6h_t:-KsyhiO3KAWBQloZ6h_u:bgw4qr0) (no related file):*
Looks like when adding the issue number to the beginning of a commit, the issue number should be followed by a colon.

---

*[src/tiles.js, line 84 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyqfgpXyAd7rY8W2c7:-KsyqfgpXyAd7rY8W2c8:b-pg3b57) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L84)):*
> ```JavaScript
>  * @return true if tileId is the id of the named tile, using a type-sensative comparison
>  */
> export function tileIsType(tileId, name) {
> ```

We could probably change `tileId` to just `id` if the variable is in the scope of a function with the word `tile` in it.

---

*[src/tiles.js, line 92 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyqrjgepBCO4hbAnUJ:-KsyqrjgepBCO4hbAnUK:b-nsa7tx) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L92)):*
> ```JavaScript
>  * @return {String} the name for the input tile id
>  */
> function getTileName(tileId) {
> ```

Change to `tileId` to `id`

---

*[src/tiles.js, line 105 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyquOjis_JfjgPqFpR:-KsyquOjis_JfjgPqFpS:b-sxg5pw) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L105)):*
> ```JavaScript
>  * @return {boolean} if the corresponding tile has a value for this property
>  */
> export function tileHasProperty(tileId, property) {
> ```

Change `tileId` to `id`

---

*[src/tiles.js, line 116 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksyr6BMukgwACuVJYGy:-Ksyr6BMukgwACuVJYGz:b-sxg5pw) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L116)):*
> ```JavaScript
>  * @return the property for the input tile
>  */
> export function getTileProperty(tileId, property) {
> ```

Change `tileId` to `id`

---

*[src/helpers/map.js, line 137 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksyrnkq8UyqjZHXtgWV:-Ksyrnkq8UyqjZHXtgWW:b-htnb02) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/helpers/map.js#L137)):*
> ```JavaScript
>         }
>       } else if (!getTileProperty(tileId, 'traversable')) {
>         generatePermanentNTSprites(x, y, mapTraversabilityCells);
> ```

Can you change the `x`'s to `xt`'s and the `y`'s to `yt`'s?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 7 of 8 files at r1, 1 of 1 files at r2.
Review status: 0 of 8 files reviewed at latest revision, 15 unresolved discussions.

---

*[src/tiles.js, line 73 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyZHouSBfcW8cBvdCo:-KsyZHouSBfcW8cBvdCp:bk7vlz3) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L73)):*
> ```JavaScript
>  * @return {(number|String)} the id for the input tile name
>  */
> function getTileId(name) {
> ```

Do we/would we ever need this except in the `tileIsType` func below?

---

*[src/tiles.js, line 82 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksy_ggDfo8PE93cSvfW:-Ksy_ggDfo8PE93cSvfX:b-f2q83m) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L82)):*
> ```JavaScript
>  * @param {number} tileID - the id for a tile
>  * @param {String} name - the name of a tile
>  * @return true if tileId is the id of the named tile, using a type-sensative comparison
> ```

sensative -> sensitive

---

*[src/tiles.js, line 84 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyYrU6m0vzDaKSFu3H:-KsyYrU6m0vzDaKSFu3I:b-vz3qe0) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L84)):*
> ```JavaScript
>  * @return true if tileId is the id of the named tile, using a type-sensative comparison
>  */
> export function tileIsType(tileId, name) {
> ```

Is this a clear name, maybe `tileIsName`?

---

*[src/tiles.js, line 95 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksy_WVfN3spZKbUkvhP:-Ksy_WVfN3spZKbUkvhQ:bijlkp7) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L95)):*
> ```JavaScript
>   assert(_.has(tileNames, tileId), `Unknown tileId: ${tileId}`);
>   const name = tileNames[tileId];
>   assert(tileIsType(tileId, name), `Input id:${tileId} was wrong data type.`);
> ```

I'm confused when this would throw, can you give an example

---

*[src/tiles.js, line 84 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyqfgpXyAd7rY8W2c7:-Ksz5LGnP74RrXFuTJZB:b32v57) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L84)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

We could probably change `tileId` to just `id` if the variable is in the scope of a function with the word `tile` in it.
</blockquote></details>

+1

---

*[src/tiles.js, line 119 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz5_dUJTqr2sQIqOW_:-Ksz5_dUJTqr2sQIqOWa:bplkk9e) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L119)):*
> ```JavaScript
>   const name = getTileName(tileId);
>   assert(tileHasProperty(tileId, property),
>     `Tile does not have property: ${name}, ${property}`);
> ```

make each param on it's own line pls

---

*[src/helpers/map.js, line 71 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz5pjm4GsP8DhFXUXa:-Ksz5pjm4GsP8DhFXUXb:b-afzk9r) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/helpers/map.js#L71)):*
> ```JavaScript
>     for (let i = 0; i < CPTL; i++) {
>       for (let j = 0; j < CPTL; j++) {
>         if (tileHasProperty(tileId, 'radius')) { // tile is CNTO
> ```

Can you make this comment on the next line and spell out CNTO

---

*[src/helpers/map.js, line 79 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz5wKkLX4AZkGpTWKZ:-Ksz5wKkLX4AZkGpTWK_:b1fxvcw) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/helpers/map.js#L79)):*
> ```JavaScript
>             gridTile[i][j] = 0;
>           }
>         // tile is noncircular and partially nontraversable (this leaves angled walls)
> ```

maybe just (angled walls)? "this leaves" is confusing to me

---

*[test/tiles.spec.js, line 13 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz6I4jq3ILrsL04yZ3:-Ksz6I4jq3ILrsL04yZ4:bq66c5m) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/test/tiles.spec.js#L13)):*
> ```JavaScript
>   getTileProperty,
>   tileHasProperty,
>   tileIsType,
> ```

why this diff

---

*[test/tiles.spec.js, line 123 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz6juCZC2qie_inCoi:-Ksz6juCZC2qie_inCoj:b2dgg2i) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/test/tiles.spec.js#L123)):*
> ```JavaScript
>   t.throws(() => { getTileProperty(-1, 'traversable'); });
>   t.throws(() => { getTileProperty('potato', 'traversable'); });
>   t.throws(() => { getTileProperty('NONEXISTANT_TILE', 'traversable'); });
> ```

nonexistant -> nonexistent

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 4 of 9 files reviewed at latest revision, 15 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyhiO3KAWBQloZ6h_t:-KszDQeDQIrIlDgte0OD:b-7ql1td) (no related file):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Looks like when adding the issue number to the beginning of a commit, the issue number should be followed by a colon.
</blockquote></details>

fixed

---

*[src/tiles.js, line 73 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyZHouSBfcW8cBvdCo:-KszDE81TJXnsC0Esd95:b-6ynxe) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L73)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Do we/would we ever need this except in the `tileIsType` func below?
</blockquote></details>

Nope. I removed the function and moved the code in

---

*[src/tiles.js, line 82 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksy_ggDfo8PE93cSvfW:-KszDck-YbzZuR3EmJk3:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L82)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

sensative -> sensitive
</blockquote></details>

Done.

---

*[src/tiles.js, line 84 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyYrU6m0vzDaKSFu3H:-Ksz7Sl5E3kEzmimSw4l:b-ox83pm) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L84)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Is this a clear name, maybe `tileIsName`?
</blockquote></details>

I'd prefer `tileHasName`. I'll make that fix

---

*[src/tiles.js, line 95 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksy_WVfN3spZKbUkvhP:-Ksz81Ra2ilqZ-tRBWJp:b-ul1jqn) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L95)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

I'm confused when this would throw, can you give an example
</blockquote></details>

If you call `getTileName('1')`. 1 is a tile, but you used the wrong type. It would also throw if you passed a decimal when you weer supposed to pass a string

---

*[src/tiles.js, line 84 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyqfgpXyAd7rY8W2c7:-KszCTgcbpV1wreJtQwb:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L84)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Done.

---

*[src/tiles.js, line 92 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyqrjgepBCO4hbAnUJ:-KszCRrZODSfUyKzdSPv:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L92)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Change to `tileId` to `id`
</blockquote></details>

Done.

---

*[src/tiles.js, line 105 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyquOjis_JfjgPqFpR:-Ksz8Ddyahdd6Bg7e_k_:bb7f2mf) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L105)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Change `tileId` to `id`
</blockquote></details>

done

---

*[src/tiles.js, line 116 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksyr6BMukgwACuVJYGy:-Ksz8EqUnMAlGGrWWssw:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L116)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Change `tileId` to `id`
</blockquote></details>

Done.

---

*[src/tiles.js, line 119 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz5_dUJTqr2sQIqOW_:-Ksz8Z5zNXXJ8l88c2SP:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/tiles.js#L119)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

make each param on it's own line pls
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 71 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz5pjm4GsP8DhFXUXa:-Ksz8qFGvNep7-EP14W_:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/helpers/map.js#L71)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Can you make this comment on the next line and spell out CNTO
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 79 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz5wKkLX4AZkGpTWKZ:-Ksz91JvHrttwRlqC8Ft:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/helpers/map.js#L79)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

maybe just (angled walls)? "this leaves" is confusing to me
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 137 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksyrnkq8UyqjZHXtgWV:-KszCPM704Wd8sUVFgU9:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/src/helpers/map.js#L137)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Can you change the `x`'s to `xt`'s and the `y`'s to `yt`'s?
</blockquote></details>

Done.

---

*[test/tiles.spec.js, line 13 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz6I4jq3ILrsL04yZ3:-KszDxrgLnFbV5KopWiJ:b-7ql1td) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/test/tiles.spec.js#L13)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

why this diff
</blockquote></details>

fixed

---

*[test/tiles.spec.js, line 123 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Ksz6juCZC2qie_inCoi:-KszEJiXIYH4PbqxKiNU:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/494046c70732f11bad0d5da1707cc2204ad656a8/test/tiles.spec.js#L123)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

nonexistant -> nonexistent
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Review status: 4 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[src/tiles.js, line 73 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KsyZHouSBfcW8cBvdCo:-KszF1n0O65u4m8N-bl3:b-5kkvq) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/8b47e4fe56edaf0edd07b1507793409f3c8442cc/src/tiles.js#L73)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Nope. I removed the function and moved the code in
</blockquote></details>

Oh shit love it :dango: 

---

*[src/tiles.js, line 13 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszLsUzyXWHn7Y4qDcu:-KszLsUzyXWHn7Y4qDcv:b-j8cwmc) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/tiles.js#L13)):*
> ```JavaScript
>  * object so that we can quickly map from an id to a name. This function depends on amBlue and
>  * amRed, so the current player must be defined. We call this function once, after our player has an
>  * id. All tile ids are stored as the data type they are in the tagpro api. 
> ```

TRAILING WHITESPACE

---

*[src/tiles.js, line 70 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszGKCaW641dGWjfFcr:-KszGKCaW641dGWjfFcs:bte4z4o) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/tiles.js#L70)):*
> ```JavaScript
> 
> /*
>  * @param {number} tileID - the id for a tile
> ```

tileID -> id

---

*[src/helpers/map.js, line 71 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszMt-kSnf9HLvR1Kio:-KszMt-kSnf9HLvR1Kip:bwurq40) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L71)):*
> ```JavaScript
>     for (let i = 0; i < CPTL; i++) {
>       for (let j = 0; j < CPTL; j++) {
>         // tile is circular non-traversable object
> ```

put an "if" here

---

*[src/helpers/map.js, line 97 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszOkEte0AeSzzIuzhG:-KszNBh4VaY5tDiZM4Cn:biav2ok) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L97)):*
> ```JavaScript
>             gridTile[i][j] = 0;
>           }
>         // tile is entirely nontraversable
> ```

blocking here until we resolve commenting convention

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 5 of 5 files at r3.
Review status: 4 of 9 files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Review status: 4 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/helpers/map.js, line 80 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-TBCSq5N_Aabe7ZdB:-Kt-TBCSq5N_Aabe7ZdC:b7spsr4) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L80)):*
> ```JavaScript
>             gridTile[i][j] = 0;
>           }
>         // tile is noncircular and partially nontraversable (angled walls)
> ```

Hmm can't really put this inline because it's referring to the next four blocks... What about putting all logic below this in it's own `else` block? Then we would have two clear blocks (circular and noncircular) and within the noncircular block have (partially and fully nontraversable)

---

*[src/helpers/map.js, line 97 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszOkEte0AeSzzIuzhG:-Kt-SxUTGeolPk6x1AHf:bfi2ij5) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L97)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

blocking here until we resolve commenting convention
</blockquote></details>

Put this back to inline

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 5 of 5 files at r3.
Review status: 4 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-V55TZTrPW3i4l5DJ:-Kt-V55TZTrPW3i4l5DK:bqo0woz) (no related file):*
`131: code refactor` is not specific enough. If a change is important enough to have it's own commit, then it deserves a precise commit message. If not, then it should be brought into an appropriate umbrella commit. In your case, I would probably rename to `131: rename tileIsType to tileHasName`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 6 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-V55TZTrPW3i4l5DJ:-Kt-uTvfS71qrTC574wG:bb7f2mf) (no related file):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`131: code refactor` is not specific enough. If a change is important enough to have it's own commit, then it deserves a precise commit message. If not, then it should be brought into an appropriate umbrella commit. In your case, I would probably rename to `131: rename tileIsType to tileHasName`
</blockquote></details>

done

---

*[src/tiles.js, line 13 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszLsUzyXWHn7Y4qDcu:-Kt-jDEwnrhCejF3y7Y3:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/tiles.js#L13)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

TRAILING WHITESPACE
</blockquote></details>

Done.

---

*[src/tiles.js, line 70 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszGKCaW641dGWjfFcr:-Kt-jKZO9vvx80OT26Kq:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/tiles.js#L70)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

tileID -> id
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 71 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszMt-kSnf9HLvR1Kio:-Kt-m1867d_x1watBet4:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L71)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

put an "if" here
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 80 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-TBCSq5N_Aabe7ZdB:-Kt-uJR5VluyuhtoQ5LS:b3ejuj4) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L80)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Hmm can't really put this inline because it's referring to the next four blocks... What about putting all logic below this in it's own `else` block? Then we would have two clear blocks (circular and noncircular) and within the noncircular block have (partially and fully nontraversable)
</blockquote></details>

I reorganized this function to make it more readable, and have less nested if else chains. Check out the style now

---

*[src/helpers/map.js, line 97 at r3](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-KszOkEte0AeSzzIuzhG:-Kt-lzgK8TrzRgvj7lQ1:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/a416477423f606f635e88ec26ba11fb0927d62e9/src/helpers/map.js#L97)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Put this back to inline
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 3 of 3 files at r4.
Review status: 6 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-V55TZTrPW3i4l5DJ:-Kt-yBh5xgvU17QvgKiI:b153ni4) (no related file):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

done
</blockquote></details>

simple present tense in last commit pls

---

*[src/helpers/map.js, line 73 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-ya24xQpo9Xq5vxnE:-Kt-ya24xQpo9Xq5vxnF:b-c5s883) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/102a3f33f5f871b8cea6e3c9ac7441974dad7b04/src/helpers/map.js#L73)):*
> ```JavaScript
>   if (
>     !tileHasProperty(tileId, 'radius') &&
>     !tileIsOneOf(tileId, ['ANGLE_WALL_1', 'ANGLE_WALL_2', 'ANGLE_WALL_3', 'ANGLE_WALL_4'])) {
> ```

put the last `) {` on a newline

---

*[src/helpers/map.js, line 80 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-zdoXTPvGG--44vKR:-Kt-zdoXTPvGG--44vKS:b-jxvxfb) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/102a3f33f5f871b8cea6e3c9ac7441974dad7b04/src/helpers/map.js#L80)):*
> ```JavaScript
>   const gridTile = init2dArray(CPTL, CPTL, 1);
> 
>   // tile is not fully traversable
> ```

can you make this more specific to this case now? "Tile is partially traversable" or something.  Maybe then the next comments can just say "tile is circular" and "tile is noncircular (angled walls)" because it'll be clear this whole next block is "partially traversable"?

---

*[src/helpers/map.js, line 81 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-yiiF1AnHz2hVtaKL:-Kt-yiiF1AnHz2hVtaKM:b60zff7) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/102a3f33f5f871b8cea6e3c9ac7441974dad7b04/src/helpers/map.js#L81)):*
> ```JavaScript
> 
>   // tile is not fully traversable
> ```

why empty line? comments should go on the line directly above the block it's explaining

---

*[src/helpers/map.js, line 85 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-z17rpXjyWFZtIvCX:-Kt-z17rpXjyWFZtIvCY:bfolo0a) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/102a3f33f5f871b8cea6e3c9ac7441974dad7b04/src/helpers/map.js#L85)):*
> ```JavaScript
>   for (let i = 0; i < CPTL; i++) {
>     for (let j = 0; j < CPTL; j++) {
>       if (tileHasProperty(tileId, 'radius')) { // if tile is circular non-traversable object
> ```

if we're doing inline, no need for "if" in the comment here, do it like you do below

---

*[src/helpers/map.js, line 95 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt002kXwamzimHigZPG:-Kt002kXwamzimHigZPH:bvxm8mz) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/102a3f33f5f871b8cea6e3c9ac7441974dad7b04/src/helpers/map.js#L95)):*
> ```JavaScript
>       } else { // tile is noncircular and partially nontraversable (angled walls)
>         // eslint-disable-next-line no-lonely-if
>         if (tileHasName(tileId, 'ANGLE_WALL_1') && j - i >= 0) {
> ```

Oooh good call de-nesting the ifs here :dango: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 6 of 9 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-V55TZTrPW3i4l5DJ:-Kt02sKHfHSg9NUg3F7m:b-8zyx4f) (no related file):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

simple present tense in last commit pls
</blockquote></details>

oops. god that habit will be touch to break. fixed

---

*[src/helpers/map.js, line 73 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-ya24xQpo9Xq5vxnE:-Kt03lKYhBRgQKxaMpWa:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/e46ada238396aa4a3d0919792e5a1059c927f793/src/helpers/map.js#L73)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

put the last `) {` on a newline
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 80 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-zdoXTPvGG--44vKR:-Kt039GcneNhU5KF8Uhr:bpzybw0) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/e46ada238396aa4a3d0919792e5a1059c927f793/src/helpers/map.js#L80)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

can you make this more specific to this case now? "Tile is partially traversable" or something.  Maybe then the next comments can just say "tile is circular" and "tile is noncircular (angled walls)" because it'll be clear this whole next block is "partially traversable"?
</blockquote></details>

Good call. Done

---

*[src/helpers/map.js, line 81 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-yiiF1AnHz2hVtaKL:-Kt03LP7A6hQhhCnlhPF:b-7ql1td) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/e46ada238396aa4a3d0919792e5a1059c927f793/src/helpers/map.js#L81)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

why empty line? comments should go on the line directly above the block it's explaining
</blockquote></details>

fixed

---

*[src/helpers/map.js, line 85 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt-z17rpXjyWFZtIvCX:-Kt03W71m_KphX4Ta2zG:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/e46ada238396aa4a3d0919792e5a1059c927f793/src/helpers/map.js#L85)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

if we're doing inline, no need for "if" in the comment here, do it like you do below
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 95 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt002kXwamzimHigZPG:-Kt03YKvvqZ8RIAuP3fm:by98f9u) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/e46ada238396aa4a3d0919792e5a1059c927f793/src/helpers/map.js#L95)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Oooh good call de-nesting the ifs here :dango: 
</blockquote></details>

Ya that was actually what forced me to refactor out and use `tileIsOneOf`, because the final else was dependent on the outer "if"s being false.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 1 of 1 files at r5.
Review status: 6 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt0677NfEWwRrkfHFAO:-Kt0677NfEWwRrkfHFAP:b-foza8) (no related file):*
Looks like travis is failing, fix tests pls

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 6 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt0677NfEWwRrkfHFAO:-Kt09qk7xbuEYF3dwPIf:b-7ql1td) (no related file):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Looks like travis is failing, fix tests pls
</blockquote></details>

fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 2 of 3 files at r4, 1 of 1 files at r5.
Review status: 6 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/tiles.js, line 116 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1agOcAphiELYcprJ2:-Kt1agOcAphiELYcprJ3:b-869di9) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/tiles.js#L116)):*
> ```JavaScript
> }
> 
> export function tileIsOneOf(id, names) {
> ```

Function level comment?

---

*[src/helpers/map.js, line 76 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1cpBNoqhouxmnnzwP:-Kt1cpBNoqhouxmnnzwQ:b-tltamk) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L76)):*
> ```JavaScript
>   ) {
>     return init2dArray(CPTL, CPTL, 0);
>   }
> ```

love it

---

*[src/helpers/map.js, line 80 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1frEBjOiUtAwcYuhp:-Kt1frEBjOiUtAwcYuhq:ba7j8r) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L80)):*
> ```JavaScript
> 
>   // tile is partially traversable
>   const gridTile = init2dArray(CPTL, CPTL, 1);
> ```

FYI

`gridTile` seems like an odd name. We could call it `tile`?

---

*[src/helpers/map.js, line 83 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1gG8MRT0xzk41zSYq:-Kt1gG8MRT0xzk41zSYr:bg9savx) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L83)):*
> ```JavaScript
>   const midCell = (CPTL - 1.0) / 2.0;
>   for (let i = 0; i < CPTL; i++) {
>     for (let j = 0; j < CPTL; j++) {
> ```

FYI

`i` and `j` could be `xc` and `yc`. If you don't want to change it here, make an issue and assign me.

---

*[src/helpers/map.js, line 102 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1bh9r-7ams_sgvd_o:-Kt1bh9r-7ams_sgvd_p:b-d57g2x) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L102)):*
> ```JavaScript
>         } else if (tileHasName(tileId, 'ANGLE_WALL_4') && i + j >= CPTL - 1) {
>           gridTile[i][j] = 0;
>         }
> ```

The :white_large_square: :space_invader: seems possibly inconsistent in this if else chain. Should the first if statement have a new line after it?

---

*[test/tiles.spec.js, line 233 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1fLm0nfN_T9JleNKc:-Kt1fLm0nfN_T9JleNKd:b5ooxg) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/test/tiles.spec.js#L233)):*
> ```JavaScript
> 
>   t.end();
> });
> ```

Add comments saying what the tile name of the given tile ID is?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 6 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/tiles.js, line 116 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1agOcAphiELYcprJ2:-Kt2TZBK0YE7tP7Xqbxv:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/tiles.js#L116)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Function level comment?
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 80 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1frEBjOiUtAwcYuhp:-Kt2Wo_89ST4C9p8IHud:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L80)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

FYI

`gridTile` seems like an odd name. We could call it `tile`?
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 83 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1gG8MRT0xzk41zSYq:-Kt2V7_0RO0le5bIoizu:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L83)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

FYI

`i` and `j` could be `xc` and `yc`. If you don't want to change it here, make an issue and assign me.
</blockquote></details>

Done.

---

*[src/helpers/map.js, line 102 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1bh9r-7ams_sgvd_o:-Kt2V_4-evoujVt504XT:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/src/helpers/map.js#L102)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

The :white_large_square: :space_invader: seems possibly inconsistent in this if else chain. Should the first if statement have a new line after it?
</blockquote></details>

Done.

---

*[test/tiles.spec.js, line 233 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt1fLm0nfN_T9JleNKc:-Kt2WpWlaFljASM7JFYU:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/c63f521a9c5117fd26a36b765d351a07fb10194f/test/tiles.spec.js#L233)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Add comments saying what the tile name of the given tile ID is?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 3 of 3 files at r6.
Review status: 6 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Review status: 6 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/helpers/map.js, line 134 at r6](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt31oqQ32J3DciIU1N0:-Kt31oqQ32J3DciIU1N1:bwpaqmt) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/9ae14c42ab670d11448b982f730a6716a7209e30/src/helpers/map.js#L134)):*
> ```JavaScript
>       if (!getTileProperty(tileId, 'permanent')) {
>         tilesToUpdate.push({ x: xt, y: yt });
>         tilesToUpdateValues.push(map[xt][yt]);
> ```

`map[xt][yt]` should be `tileId`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 3 of 3 files at r6.
Review status: 6 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/helpers/map.js, line 134 at r6](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188#-Kt31oqQ32J3DciIU1N0:-Kt52BcRJBxuGhnwLP2D:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/9ae14c42ab670d11448b982f730a6716a7209e30/src/helpers/map.js#L134)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`map[xt][yt]` should be `tileId`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 1 of 1 files at r7.
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 1 of 1 files at r7.
Review status: 8 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/188)*
<!-- Sent from Reviewable.io -->
